proxy/src/registry.rs

```diff
@@ -403,21 +399,24 @@ impl Client for Registry {
                 fee,
             },
         );
-        // TODO(xla): Unpack the result to find out if the application of the transaction failed.
-        let register_applied = self.client.submit_transaction(register_tx).await?.await?;
+        let applied = self.client.submit_transaction(register_tx).await?.await?;
+        applied.result?;
+
         let tx = Transaction {
-            id: Hash(register_applied.tx_hash),
-            messages: vec![Message::OrgRegistration {
-                id: Id(org_id.clone()),
-            }],
+            id: Hash(applied.tx_hash),
+            messages: vec![Message::OrgRegistration { id: org_id.clone() }],
             state: transaction::State::Applied {
-                block: Hash(register_applied.block),
+                block: Hash(applied.block),
             },
             timestamp: SystemTime::now(),
         };
 
         // TODO(xla): Remove autmoatic prepayment once we have proper balances.
-        let org = self.client.get_org(org_id).await?.expect("org not present");
+        let org = self
+            .client
+            .get_org(org_id.0)
```

Comment:
This is a bit ugly -- it's weird that some functions don't expect the `Id`, and others do.

Reply:
This is a method on the underyling client not on our own implementation.

Reply:
Since you don't control the API, perhaps having an `into()` that does what you want is cleaner, since it's a one-way function, vs. giving full access to the inner value.

Reply:
Or `as_str()` if that's what you need.

Reply:
You can then make the inner value non-`pub`

Reply:
As these newtypes are just stop-gap solutions until we have reflected these concerns upstream in radicle-registry I will hold off to improve the ergonomics for now. Valid input tho and thanks for pointing it out.